### DISCUSSION TITLE
How to host serverless function in src/api/ and not in api/ (Gatsby)

### DISCUSSION THREAD

**Discussion by don-esteban** [vercel/community#571] [Help] [answered]
Hi all,

I have a serverless function which deploys nicely on Gatsby Cloud. The standard source directory is

`src/api/`

Vercel uses

`api/`

How can I change this to `src/api/`?

Thanks
Stefan

**Comment by samcx** [comment, marked as answer]
Unfortunately, we do not have something like this Gatsby (similar to Next.js's `/pages/api`). Is it not possible to create a Serverless Function via `/api`?

**Reply by don-esteban** [reply]
Thanks for your reply. I'm looking for a way to host our site at Gatsby Cloud and Vercel the same time. Due to experiences with outages (we can't afford two or three hours downtime, which happened almost monthly), we thought about switching DNS in the case of problems. That's not a very clever solution, but the only thing we came up until now.

**Reply by samcx** [reply]
It's possible, but since we don't support `src/api` with Gatsby at the moment, it'll be very cumbersome to maintain two almost identical apps.

I recommend moving your Serverless Functions outside to the root `/api` and hosting on Vercel, if possible!
